### PR TITLE
Fix nonce validation logic - Closes #5983

### LINKED ISSUE #5983: Wrong logic for nonce validation on transaction.create
### Expected behavior

When using an account with nonce === 0 transaction.create should not fail.

### Actual behavior

It's failing as a check is implemented in a way that a truthy value is possible.

### Steps to reproduce

create a transaction for an account with nonce === 0 using lisk-api-client

### Which version(s) does this affect? (Environment, OS, etc...)

5.1

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #5983 

### How was it solved?

* Fix logic to check for undefined, null or negative nonce values in the input

### How was it tested?

* Add unit test for cases when nonce is zero, negative & undefined

### CODE REVIEW

File: elements/lisk-api-client/test/unit/transaction.spec.ts [hunk 1]
```diff
@@ -121,14 +121,50 @@ describe('transaction', () => {
 				});
 			});
 
-			describe('when called without nonce in input', () => {
+			describe('when called without nonce in input and account is not supporting nonce', () => {
```
Comment (pablitovicente) [inline elements/lisk-api-client/test/unit/transaction.spec.ts:124]: Maybe try to improve this phrase it's confusing.

**Review by pablitovicente — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]
